brownfield_django/main/views.py

```diff
@@ -42,18 +35,41 @@ def get_queryset(self):
         filtering against the request.user
         excluding against an `exclude_username` query parameter in the URL.
         """
-        if self.request.user.profile.is_student():
-            return Course.objects.none()
-
-        queryset = Course.objects.filter(archive=False).order_by('name')
+        queryset = Course.objects.none()
 
+        if self.request.user.profile.is_student():
+            queryset = Course.objects.none()
         if self.request.user.profile.is_teacher():
-            return queryset.filter(
+            queryset = Course.objects.filter(
                 professor=self.request.user).order_by('name')
-
         if self.request.user.profile.is_admin():
-            return queryset
+            queryset = Course.objects.filter(archive=False).order_by('name')
+        return queryset
+
+
+class DocumentViewSet(viewsets.ModelViewSet):
+    queryset = Document.objects.all()
+    serializer_class = DocumentSerializer
+
+    def get_queryset(self):
+        '''
+        Form Docs: queryset that should be used for list views,
+        and that should be used as the base for lookups in detail views.
+        '''
+        course_pk = self.request.QUERY_PARAMS.get('course', None)
+        doc_pk = self.kwargs.get('pk', None)
+        up = self.request.user.profile
+        queryset = Document.objects.none()
 
+        if up.is_student():
+            queryset = Document.objects.none()
+        elif up.is_admin() or up.is_teacher():
+            if course_pk is not None:
+                queryset = Document.objects.filter(course__pk=course_pk)
+            if doc_pk is not None:
```

Comment:
For this app yeah there are students admins and teachers, but I thought it would be better to explicitly make sure that they are an admin or teacher before sending back all the documents